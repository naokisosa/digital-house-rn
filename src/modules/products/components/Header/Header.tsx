import { StyleSheet, View } from 'react-native';
import React from 'react';
import { BaseText } from '@components';
import { colors } from '@style';

interface Props {
  title?: string;
}

const Header = ({ title }: Props) => {
  return (
    <View style={styles.header}>
      <BaseText style={styles.title}>{title}</BaseText>
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  header: {
    height: 150,
    backgroundColor: colors.secondary,
    justifyContent: 'flex-end',
    paddingHorizontal: 20,
  },
  title: {
    fontWeight: '800',
    fontSize: 24,
    marginBottom: 24,
  },
});
