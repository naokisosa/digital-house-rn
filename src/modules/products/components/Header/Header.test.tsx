import 'react-native';
import React from 'react';

import Header from './Header';
import renderer from 'react-test-renderer';
import { SINGLE_PRODUCT } from '@mocks';

describe('Test Header component', () => {
  it('renders correctly with title', () => {
    const { toJSON } = renderer.create(<Header title={SINGLE_PRODUCT.product} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
