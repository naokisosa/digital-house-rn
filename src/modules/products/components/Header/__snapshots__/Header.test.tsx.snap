// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header component renders correctly with title 1`] = `
<View
  style={
    {
      "backgroundColor": "#CFD6FF",
      "height": 150,
      "justifyContent": "flex-end",
      "paddingHorizontal": 20,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontFamily": "Avenir",
        },
        {
          "fontSize": 24,
          "fontWeight": "800",
          "marginBottom": 24,
        },
      ]
    }
  >
    Recycled Wooden Salad
  </Text>
</View>
`;
